test/e2e/tls_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware-tanzu/antrea/pkg/apis"
+)
+
+const (
+	cipherSuite    = tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 // a TLS1.2 Cipher Suite
+	cipherSuiteStr = "ECDHE-RSA-AES128-GCM-SHA256"
+)
+
+var (
+	cipherSuites             = []uint16{cipherSuite}
+	opensslTLS13CipherSuites = []string{"TLS_AES_128_GCM_SHA256", "TLS_AES_256_GCM_SHA384", "TLS_CHACHA20_POLY1305_SHA256"}
+)
+
+// TestTLSCipherSuites tests Cipher Suite and TLSVersion config on Antrea Apiserver, Controller side or Agent side.
+func TestTLSCipherSuites(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	data.configureTLS(t, cipherSuites, "VersionTLS12")
+
+	controllerPod, err := data.getAntreaController()
+	assert.NoError(t, err, "failed to get Antrea Controller Pod")
+	controllerPodName := controllerPod.Name
+	controlPlaneNode := controlPlaneNodeName()
+	agentPodName, err := data.getAntreaPodOnNode(controlPlaneNode)
+	assert.NoError(t, err, "failed to get Antrea Agent Pod Name on Control Plane Node")
+
+	tests := []struct {
+		podName       string
+		containerName string
+		apiserver     int
+		apiserverStr  string
+	}{
+		{controllerPodName, controllerContainerName, apis.AntreaControllerAPIPort, "Controller"},
+		{agentPodName, agentContainerName, apis.AntreaAgentAPIPort, "Agent"},
+	}
+	for _, tc := range tests {
+		data.checkTLS(t, tc.podName, tc.containerName, tc.apiserver, tc.apiserverStr)
```

Comment:
It could create a subtest with the specified name to include context to the output of the test log. For example:
```
	tests := []struct {
                name          string
		podName       string
		containerName string
		apiserver     int
	}{
		{"Controller", controllerPodName, controllerContainerName, apis.AntreaControllerAPIPort},
		{"Agent", agentPodName, agentContainerName, apis.AntreaAgentAPIPort},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
		        data.checkTLS(t, tc.podName, tc.containerName, tc.apiserver)
```

Reply:
Good idea. Updated.